app/src/main/java/com/example/euriskocodechallenge/common/ImageUtils.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.euriskocodechallenge.common
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.graphics.ImageDecoder
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+import android.util.Base64
+import com.example.euriskocodechallenge.R
+import java.io.ByteArrayOutputStream
+
+
+fun Uri.toBitmap(context: Context): Bitmap {
+
+    try {
+
+        return if (Build.VERSION.SDK_INT < 28) {
+            MediaStore.Images.Media.getBitmap(
+                context.contentResolver,
+                this
+            )
+        } else {
+            val source = ImageDecoder.createSource(context.contentResolver, this)
+            source.let { ImageDecoder.decodeBitmap(it) }
+        }
+
+
+    } catch (e: Exception) {
+
+        e.printStackTrace()
```

Comment:
@martin371998 avoid printing exceptions, instead create a debug logger that prints logs only on Debug builds